vignettes/lab_grading.Rmd

```diff
@@ -0,0 +1,395 @@
+namely `ATOXGR`. Grades held in `ATOXGRL` will be given a negative value in `ATOXGR`
+to distinguish between low and high values.
+
+```{r, eval=TRUE}
+adlb <- adlb %>% 
+  derive_var_atoxgr()
+```
+
+```{r, eval=TRUE, echo=FALSE}
+dataset_vignette(
+  atoxgr_criteria_ctcv4,
+  display_vars = vars(TERM, SI_UNIT_CHECK),
+  filter = !is.na(SI_UNIT_CHECK)
+)
+```
+
+
+# NCI-CTCAEV4 implementation
+
+## Terms graded
+
+Grading is implemented for those lab tests where a lab value is included in the grading definition,
+{admiral} does NOT try to read any other data to determine the grade, and only the ADLB VAD is used.
+The following CTCAE v4.0 SOC values were identified for grading, these are “Investigations", 
+“Metabolism and nutrition disorders” and “Blood and lymphatic system disorders”.
+
+From these SOC values the following terms criteria is implemented in {admiral} 
+
+From SOC = “Investigations" there are 21 CTCAE v4.0 Terms:
+
+  + Activated partial thromboplastin time prolonged
+  + Alanine aminotransferase increased
+  + Alkaline phosphatase increased
+  + Aspartate aminotransferase increased
+  + Blood bilirubin increased
+  + CD4 lymphocytes decreased
+  + Cholesterol high
+  + CPK increased
+  + Creatinine increased
+  + Fibrinogen decreased
+  + GGT increased
+  + Haptoglobin decreased
+  + Hemoglobin increased
+  + INR increased
+  + Lipase increased
+  + Lymphocyte count decreased
+  + Lymphocyte count increased
+  + Neutrophil count decreased
+  + Platelet count decreased
+  + Serum amylase increased
+  + White blood cell decreased
+
+From the SOC =  “Metabolism and nutrition disorders” there are 14 CTCAE v4.0 Terms:
+
+  + Hypercalcemia
+  + Hyperglycemia
+  + Hyperkalemia
+  + Hypermagnesemia
+  + Hypernatremia
+  + Hypertriglyceridemia
+  + Hyperuricemia
+  + Hypoalbuminemia
+  + Hypocalcemia
+  + Hypoglycemia
+  + Hypokalemia
+  + Hypomagnesemia
+  + Hyponatremia
+  + Hypophosphatemia
+
+From the SOC =  “Blood and lymphatic system disorders” there are 2 CTCAE v4.0 Terms:
+
+  + Anemia
+  + Leukocytosis
+
+## Updates made to TERM
+
+For terms "Hypocalcemia" and "Hypercalcemia" the criteria is provided for Calcium and Ionized Calcium,
+therefore {admiral} created a row for each in the metadata, this is noted in the COMMENT variable of
+the metadata:
+
+```{r, eval=TRUE, echo=FALSE}
+dataset_vignette(
+  atoxgr_criteria_ctcv4,
+  display_vars = vars(TERM, COMMENT),
+  filter = str_detect(TERM, 'calcemia')
+)
+```
+
+Similarly, there is criteria applicable to Fasting Glucose as well as non-Fasting Glucose for "Hyperglycemia"
+so again this was split into 2 rows, and noted in the COMMENT variable. Note "Hypoglycemia" does not require to
+be split into 2 rows:
+
+```{r, eval=TRUE, echo=FALSE}
+dataset_vignette(
+  atoxgr_criteria_ctcv4,
+  display_vars = vars(TERM, COMMENT),
+  filter = str_detect(TERM, 'glycemia')
+)
+```
+
+## Assumptions made when grading
+
```

Comment:
(Just curious, is this the final strategy of handling grading that involve information outside of LB?)

Reply:
Not sure what you mean? do you have something in mind?

Reply:
Sorry for the confusion, I meant for some grading like the “Hyperuricemia” example below, assumptions are made and documented in the comment column. If users do not agree with the assumption, they’ll just update the code stored in the metadata? (Not an urgent question as we are not updating anything for this release). Thanks!